server/embed/config_logging.go

```diff
@@ -78,7 +80,19 @@ func (cfg *Config) setupLogging() error {
 				}
 			}
 		}
-
+		if cfg.LogRotate {
+			zap.RegisterSink("rotate", func(u *url.URL) (zap.Sink, error) {
+				return rotationSink{
+					Logger: &lumberjack.Logger{
```

Comment:
As the lumberjeck config:  is JSON marshallable: 
  https://github.com/natefinch/lumberjack/blob/47ffae23317c5951a2a6267a069cf676edf53eb6/lumberjack.go#L83,

maybe instead being opinionated about which flags to we support in etcd, we should just have a single:
  "--log_rotation_lumberjack_json={...}"

Reply:
Yeah I like that +1